packages/repository/src/mixins/repository.mixin.ts

```diff
@@ -184,17 +180,30 @@ export function RepositoryMixin<T extends Class<any>>(superClass: T) {
      * @param options Migration options, e.g. whether to update tables
      * preserving data or rebuild everything from scratch.
      */
-    async migrateSchema(options?: SchemaMigrationOptions): Promise<void> {
-      const repoBindings = this.findByTag('repository');
+    async migrateSchema(options: SchemaMigrationOptions = {}): Promise<void> {
+      const operation = options.dropExistingSchema
+        ? 'automigrate'
+        : 'autoupdate';
 
-      for (const b of repoBindings) {
-        const repo = await this.get(b.key);
+      // Instantiate all repositories to ensure models are registered & attached
+      // to their datasources
+      const repoBindings: Readonly<Binding<unknown>>[] = this.findByTag(
+        'repository',
+      );
+      await Promise.all(repoBindings.map(b => this.get(b.key)));
```

Comment:
I'm wondering how this will affect performance if we're loading up all the repository instances? Should we do some sort of filtering now that we support select models in the `migrateSchema` function?

Reply:
Great question!

We are creating instances of repository classes. This is a lightweight task - a repository instance is created and a new PersistedModel class is registered with the appropriate datasource. I am not anticipating any performance issues. 

Migration is a one-time action so even if its performance is not optimal, the affect on user applications should be minimal.

To allow datasource migration to define foreign key constraints, it may be necessary to setup related models even if they are not migrated. For example, if we are migrating only an Order model which has a belongs to relation to User, then we should instantiate UserRepository too, so that the datasource/connector knows about both sides of the relation.

Finally, I am not aware of a reasonably easy & sufficiently robust way of how to filter repositories based on model names, because the repository binding does not provide any information about the model it is providing data access to.

Reply:
Thank you for clarifying 💯!